testbed/core/views.py

```diff
@@ -72,3 +82,200 @@ def index(request):
         'destination_actor': user_actors.filter(role=Actor.ROLE_DESTINATION).first(),
         'oauth_form': oauth_form,
     })
+
+
+# OAuth Testing Views
+
+def oauth_callback(request):
+    """
+    Handle the callback from the OAuth server
+    This displays the authorization code or error message
+    """
+    # Get query parameters
+    code = request.GET.get('code')
+    error = request.GET.get('error')
+    error_description = request.GET.get('error_description')
+    state = request.GET.get('state')
+    
+    # Initialize context with received parameters
+    context = {
+        'code': code,
+        'error': error,
```

Comment:
Where would this error come from ? The callback URL is put together with parameters by the resource server in the OAuth dance, so what kinds of errors would they put on the URL? 

If an error comes in from the resource server in the URL parameter, we should probably do two things:
* Log that we got an error from the resource server - and log that we should probably understand why we got an error from the resource server
* Go straight to render, without trying to validate the state parameter - this is an error to report to user
* comment where this error comes from for the next time we look at this code